web3/contract.py

```diff
@@ -497,11 +538,11 @@ def _encode_constructor_data(cls, args=None, kwargs=None):
 
             deploy_data = to_hex(cls.bytecode)
 
-        return deploy_data
+        return HexStr(deploy_data)
 
 
-def mk_collision_prop(fn_name):
-    def collision_fn():
+def mk_collision_prop(fn_name: str) -> Callable[[], None]:
```

Comment:
Ahh that's cool, wasn't aware of that type - seems like I should be using that in a lot more places rather than `-> None:`? And using `-> None:` only when `None`  is being explicitly returned?